sdk/src/types/block/semantic/mod.rs

```diff
@@ -281,7 +286,15 @@ impl<'a> SemanticValidationContext<'a> {
                         TransactionFailureReason::TransactionCapabilityManaBurningNotAllowed,
                     ));
                 }
-            } else {
+            // If mana rewards were provided or none of the inputs can claim rewards
+            // then the mismatch should return an error. Otherwise the mismatch may
+            // still be acceptable.
+            } else if self.mana_rewards.is_some()
+                || !self
+                    .input_chains
+                    .iter()
+                    .any(|(chain_id, input)| input.can_claim_rewards(self.output_chains.get(chain_id).copied()))
+            {
```

Comment:
Maybe if you keep the rewards Hashmap from `Selected` this would be a little bit more defined/verified and we could move the code to the transitions where I think it should be

Reply:
Would you expect this to still allow the transaction to pass validation if there is no reward provided for an input and the amounts don't match? Or should there be a new error to tell the user that they may need to provide rewards for the input?

Reply:
I think I would error, just an extra layer of verification, what do you think ?